ogre/src/OgreLidarVisual.cc

```diff
@@ -171,77 +212,109 @@ void OgreLidarVisual::Update()
+          this->dataPtr->deadZoneRayFans.push_back(line);
+          this->dataPtr->deadZoneRayFans[j]->AddPoint(
+                      ignition::math::Vector3d::Zero);
+
+          line = std::shared_ptr<OgreDynamicLines>(
+                      new OgreDynamicLines(MT_TRIANGLE_STRIP));
+
+          #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
+            line->setMaterial("Lidar/BlueStrips");
+          #else
+            mat = Ogre::MaterialManager::getSingleton().getByName(
+                                                    "Lidar/BlueStrips");
+            line->setMaterial(mat);
+          #endif
+          mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+          this->Node()->attachObject(mv.get());
+          this->dataPtr->rayStrips.push_back(line);
+        }
+      }
+    }
 
-      this->Node()->attachObject(mv.get());
-      this->dataPtr->rayStrips.push_back(line);
-
-      line = std::shared_ptr<OgreDynamicLines>(
-                  new OgreDynamicLines(MT_TRIANGLE_STRIP));
-
-      #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
-        line->setMaterial("Lidar/LightBlueStrips");
-      #else
-        mat = Ogre::MaterialManager::getSingleton().getByName(
-                                                "Lidar/LightBlueStrips");
-        line->setMaterial(mat);
-      #endif
-      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
-      this->Node()->attachObject(mv.get());
-      this->dataPtr->noHitRayStrips.push_back(line);
-
-      line = std::shared_ptr<OgreDynamicLines>(
-                  new OgreDynamicLines(MT_TRIANGLE_FAN));
-
-      #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
-        line->setMaterial("Lidar/TransBlack");
-      #else
-        mat = Ogre::MaterialManager::getSingleton().getByName(
-                                                "Lidar/TransBlack");
-        line->setMaterial(mat);
-      #endif
-      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
-      this->Node()->attachObject(mv.get());
-      this->dataPtr->deadZoneRayFans.push_back(line);
-      this->dataPtr->deadZoneRayFans[j]->AddPoint(
-                  ignition::math::Vector3d::Zero);
-
-
-      line = std::shared_ptr<OgreDynamicLines>(
-                  new OgreDynamicLines(MT_LINE_LIST));
-
-      #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
-        line->setMaterial("Lidar/BlueRay");
-      #else
-        mat = Ogre::MaterialManager::getSingleton().getByName(
+    else if (this->dataPtr->lidarVisType ==
+                  LidarVisualType::LVT_POINTS)
+    {
+      if (j+1 > this->dataPtr->points.size())
+      {
+        std::shared_ptr<OgreDynamicLines> line =
+                  std::shared_ptr<OgreDynamicLines>(
+                              new OgreDynamicLines(MT_POINTS));
+
+        #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
+              line->setMaterial("Lidar/BlueRay");
+        #else
+            mat = Ogre::MaterialManager::getSingleton().getByName(
```

Comment:
another one

Reply:
@iche033 I think only this one needs the declaration. The others above are in the same namespace so the variable can be reused. I've pushed the edit for the same.

Reply:
[new build](https://build.osrfoundation.org/job/ign_rendering-pr-win/858/) looks good!